cl/citations/match_citations_queries.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+
+from django_elasticsearch_dsl.search import Search
+from elasticsearch_dsl import Q
+from elasticsearch_dsl.query import Query
+from elasticsearch_dsl.response import Hit, Response
+from eyecite.models import FullCaseCitation
+
+from cl.citations.types import SupportedCitationType
+from cl.citations.utils import (
+    QUERY_LENGTH,
+    get_years_from_reporter,
+    make_name_param,
+)
+from cl.search.documents import OpinionDocument
+from cl.search.models import Opinion
+
+
+def fetch_citations(search_query: Search) -> list[Hit]:
+    """Fetches all citation matches from Elasticsearch based on the provided
+    search query.
+
+    :param search_query: The Elasticsearch DSL Search object.
+    :return: A list of ES Hits objects.
+    """
```

Comment:
I believe we should update this docstring to explicitly state that the method returns the top 2 matching documents. This will enhance understanding for future users.

Reply:
Sure, I've updated the docstrings and explain why a size of 2.